### PR TITLE
 Populate the controls for the OCP4 STIG

### PR DESCRIPTION
**Pull request by DhritiShikhar**
#### Description:

-  Used `utils/build_stig_control.py` to bootstrap the controls structure for OCP4
-  Creates two seperate sub directories - one for ocp4 and one for rhcos as mentioned [here](https://issues.redhat.com/browse/CMP-1206?focusedCommentId=19706831&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-19706831)

#### Rationale:

- Fixes https://issues.redhat.com/browse/CMP-1206

### CODE REVIEW

**Review by Mab879 — CHANGES_REQUESTED** [review]
You can remove the `controls` key and its children as this is duplicative of the data in the files.

**Comment by DhritiShikhar** [comment]
@Mab879 Removed `controls` key

File: controls/ocp4/stig_control.yml [hunk 1]
```diff
@@ -0,0 +1,1050 @@
+policy: Container Platform Security Requirements Guide
```
[the remark this answers is not in the document]
Comment (DhritiShikhar) [reply controls/ocp4/stig_control.yml:1]: Done

**Comment by jhrozek** [comment]
> Please add the XML you used to create this control into `shared/references`. My suggestion for the filename would be `disa-ctr-srg-v1r3.xml`.

Curious, how did you generate the file?

**Comment by DhritiShikhar** [comment]
> Curious, how did you generate the file?

@jhrozek Matthew pointed me to this XCCDF file on  https://public.cyber.mil/stigs/downloads

**Comment by jhrozek** [comment]
general question, do you also plan on submitting RHCOS4 controls?

**Comment by DhritiShikhar** [comment]
@jhrozek Not sure. https://issues.redhat.com/browse/CMP-1206 talks just about OCP4 controls.
May be @Mab879 can suggest if RHCOS4 controls are also required or not.

**Comment by Mab879** [comment]
After looking at your control files, I believe that starting over with the following command should give a better start.

```bash
./utils/build_stig_control.py -s -m shared/references/disa-ctr-srg-v1r3.xml -o controls/stig_ctr.yml -p ocp4 -ref srg
```

1. This will get you control files with rules
2. This will standardize location based on the General Purpose Operating System (GPOS) SRG

File: controls/stig_ctr/SRG-APP-000014-CTR-000035.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+controls:
+- id: SRG-APP-000014-CTR-000035
+  levels:
+  - medium
+  title: The container platform must use TLS 1.2 or greater for secure container image
```
Comment (Mab879) [inline controls/stig_ctr/SRG-APP-000014-CTR-000035.yml:5]: This won't hold up the PR, but replacing `The container platform` with `{{{ full_name }}}` might be helpful in the future.

File: controls/stig_ctr.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+policy: Container Platform Security Requirements Guide
+title: Container Platform Security Requirements Guide
+id: stig_ocp4
```
Comment (Mab879) [inline controls/stig_ctr.yml:3]:
```suggestion
id: srg_ctr
```

It looks like I made a small typo when giving you that command. The id should be `srg_ctr`. So the files and folder need to be moved, sorry about that.

**Review by Mab879 — CHANGES_REQUESTED** [review]

File: controls/stig_ctr.yml (hunk 3 above)
Comment (DhritiShikhar) [reply controls/stig_ctr.yml:3]: No worries. On it.

**Comment by jhrozek** [comment]
On Mon, May 16, 2022 at 01:01:40PM -0700, Dhriti Shikhar wrote:
> @jhrozek Not sure. https://issues.redhat.com/browse/CMP-1206 talks just about OCP4 controls.
> May be @Mab879 can suggest if RHCOS4 controls are also required or not.

The overall idea is to get rid of using spreadsheets as the source of
truth (you can't actually use them and run CI on them..) but go the
other way around, use CaC as the source of truth and generate the
spreadsheets (which DISA operates on) from CaC.

So yes, we also need the RHCOS controls. A separate PR is definitely
better, also because I suspect that many more rules would be tagged with
SRG IDs under linux_os than under applications/openshift and we need to
cross-check what is generated with what is in the sheet we had submitted
to DISA earlier.

**Pull request merged**